File: generate_icons.py
"""
Generate PNG icons for the YT Analytics Chrome Extension.
Run: python generate_icons.py
Requires: Pillow  (pip install Pillow)
"""
from PIL import Image, ImageDraw
import os, math

sizes = [16, 48, 128]
icons_dir = os.path.join(os.path.dirname(__file__), "icons")
os.makedirs(icons_dir, exist_ok=True)

def draw_rounded_rect(draw, xy, radius, fill):
    x0, y0, x1, y1 = xy
    draw.rectangle([x0 + radius, y0, x1 - radius, y1], fill=fill)
    draw.rectangle([x0, y0 + radius, x1, y1 - radius], fill=fill)
    draw.ellipse([x0, y0, x0 + 2*radius, y0 + 2*radius], fill=fill)
    draw.ellipse([x1 - 2*radius, y0, x1, y0 + 2*radius], fill=fill)
    draw.ellipse([x0, y1 - 2*radius, x0 + 2*radius, y1], fill=fill)
    draw.ellipse([x1 - 2*radius, y1 - 2*radius, x1, y1], fill=fill)

for size in sizes:
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Dark background
    bg_r = max(2, int(size * 0.18))
    draw_rounded_rect(draw, (0, 0, size, size), bg_r, (15, 15, 17, 255))

    # Red YouTube badge
    pad    = max(1, int(size * 0.1))
    yt_y0  = int(size * 0.28)
    yt_y1  = int(size * 0.72)
    yt_r   = max(2, int(size * 0.1))
    draw_rounded_rect(draw, (pad, yt_y0, size - pad, yt_y1), yt_r, (255, 0, 0, 255))

    # White triangle
    cx, cy = size // 2, size // 2
    h      = int(size * 0.3)
    w      = int(size * 0.26)
    tx     = cx - int(w * 0.35)
    triangle = [
        (tx,         cy - h // 2),
        (tx + w,     cy),
        (tx,         cy + h // 2),
    ]
    draw.polygon(triangle, fill=(255, 255, 255, 255))

    out_path = os.path.join(icons_dir, f"icon{size}.png")
    img.save(out_path, "PNG")
    print(f"Saved {out_path}")

print("All icons generated!")

File: generate_icons.js
// Simple Node script to generate PNG icons from canvas
// Run with: node generate_icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

const sizes = [16, 48, 128];

function drawIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Background rounded rect
  const r = size * 0.2;
  ctx.fillStyle = '#0f0f11';
  ctx.beginPath();
  ctx.moveTo(r, 0);
  ctx.lineTo(size - r, 0);
  ctx.quadraticCurveTo(size, 0, size, r);
  ctx.lineTo(size, size - r);
  ctx.quadraticCurveTo(size, size, size - r, size);
  ctx.lineTo(r, size);
  ctx.quadraticCurveTo(0, size, 0, size - r);
  ctx.lineTo(0, r);
  ctx.quadraticCurveTo(0, 0, r, 0);
  ctx.closePath();
  ctx.fill();

  // YouTube red rounded rect
  const rr = size * 0.14;
  const rrX = size * 0.1, rrY = size * 0.28;
  const rrW = size * 0.8, rrH = size * 0.44;
  ctx.fillStyle = '#FF0000';
  ctx.beginPath();
  ctx.moveTo(rrX + rr, rrY);
  ctx.lineTo(rrX + rrW - rr, rrY);
  ctx.quadraticCurveTo(rrX + rrW, rrY, rrX + rrW, rrY + rr);
  ctx.lineTo(rrX + rrW, rrY + rrH - rr);
  ctx.quadraticCurveTo(rrX + rrW, rrY + rrH, rrX + rrW - rr, rrY + rrH);
  ctx.lineTo(rrX + rr, rrY + rrH);
  ctx.quadraticCurveTo(rrX, rrY + rrH, rrX, rrY + rrH - rr);
  ctx.lineTo(rrX, rrY + rr);
  ctx.quadraticCurveTo(rrX, rrY, rrX + rr, rrY);
  ctx.closePath();
  ctx.fill();

  // White play triangle
  const cx = size / 2, cy = size / 2;
  const th = size * 0.26;
  const tw = size * 0.23;
  ctx.fillStyle = '#FFFFFF';
  ctx.beginPath();
  ctx.moveTo(cx - tw * 0.4, cy - th / 2);
  ctx.lineTo(cx + tw * 0.8, cy);
  ctx.lineTo(cx - tw * 0.4, cy + th / 2);
  ctx.closePath();
  ctx.fill();

  return canvas.toBuffer('image/png');
}

const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) fs.mkdirSync(iconsDir);

for (const size of sizes) {
  const buf = drawIcon(size);
  fs.writeFileSync(path.join(iconsDir, `icon${size}.png`), buf);
  console.log(`Generated icon${size}.png`);
}
console.log('Done!');
